README.md

```diff
@@ -265,6 +265,154 @@ end)
+		canContinue = true
+		return
+	end
+
+	if doorData.configfile == '' then doorData.configfile = false end
+	if doorData.job == '' then doorData.job = false end
+	if doorData.gang == '' then doorData.gang = false end
+	if doorData.cid == '' then doorData.cid = false end
+	if doorData.item == '' then doorData.item = false end
+	if doorData.doorlabel == '' then doorData.doorlabel = nil end
+	if doorData.pickable ~= 'true' then doorData.pickable = nil end
+	if doorData.cantunlock ~= 'true' then doorData.cantunlock = nil end
+	if doorData.hidelabel ~= 'true' then doorData.hidelabel = nil end
+
+	doorData.locked = 'true'
+	doorData.distance = tonumber(doorData.distance)
+	if doorData.doortype == 'door' or doorData.doortype == 'sliding' or doorData.doortype == 'garage' then
+		local heading, result, entityHit
+		local entity, coords, model= 0, 0, 0
+		QBCore.Functions.Notify("Point any Weapon at the Door. Press [E] to confirm it.", 'primary', 30000)
+		while true do
+			if IsPlayerFreeAiming(PlayerId()) then
+				result, entityHit = raycastWeapon()
+				if result and entityHit ~= entity then
+					SetEntityDrawOutline(entity, false)
+					SetEntityDrawOutline(entityHit, true)
+					entity = entityHit
+					coords = GetEntityCoords(entity)
+					model = GetEntityModel(entity)
+					heading = GetEntityHeading(entity)
+				end
+				if entity and IsControlPressed(0, 38) then break end
+			end
+			Wait(0)
+		end
+		SetEntityDrawOutline(entity, false)
+		if not model or model == 0 then QBCore.Functions.Notify(Lang:t("error.door_not_found"), 'error') canContinue = true return end
+		result = DoorSystemFindExistingDoor(coords.x, coords.y, coords.z, model)
+		if result then QBCore.Functions.Notify(Lang:t("error.door_registered"), 'error') canContinue = true return end
+		doorData.doorHash = 'door_'..doorData.dooridentifier
+		AddDoorToSystem(doorData.doorHash, model, coords, false, false, false)
+		DoorSystemSetDoorState(doorData.doorHash, 4, false, false)
+		coords = GetEntityCoords(entity)
+		heading = GetEntityHeading(entity)
+		RemoveDoorFromSystem(doorData.doorHash)
+		doorData.entity = entity
+		doorData.coords = coords
+		doorData.model = model
+		doorData.heading = heading
+		TriggerServerEvent('qb-doorlock:server:saveNewDoor', doorData, false)
+		canContinue = true
+	else
+		local result, entityHit
+		local entity, coords, heading, model = {0, 0}, {0, 0}, {0, 0}, {0, 0}
+		for i = 1, 2 do
+			QBCore.Functions.Notify("Point any Weapon at the Door. Press [E] to confirm it.", 'primary', 30000)
+			while true do
+				if IsPlayerFreeAiming(PlayerId()) then
+					result, entityHit = raycastWeapon()
+					if result and entityHit ~= entity[i] then
+						SetEntityDrawOutline(entity[i], false)
+						SetEntityDrawOutline(entityHit, true)
+						entity[i] = entityHit
+						coords[i] = GetEntityCoords(entity[i])
+						model[i] = GetEntityModel(entity[i])
+						heading[i] = GetEntityHeading(entity[i])
```

Comment:
all of this stuff looks like duplicate code from the 342 to 347 block, put it all behind a function so you dont have to duplicate code

Reply:
there is a difference between these 2 i think